terraform.md

```diff
@@ -50,14 +50,41 @@ variable "web_cidrs" {
 }
 ```
 
-### Defaults
-
+### Value Assignment
+Values can be assigned to variables in Environment 'vars' files, variable block 'defaults' or 'locals' blocks. Ensure the most appropriate method is used for each definition to aid readability and avoid duplication.  
+Some guidance:
+#### Defaults
 If it makes sense to do so then default values should be provided. Defaults are definitely preferable to duplicated values between environments
+#### Environment vars
+Set values here only for variables that will change per environment.
+#### Locals Blocks
+Set values here to avoid repeating constants inside a module, and where the context is local to the module. Usually there will be just one locals block in a module, but you can add more to separate variable definitions if this aides the readability of your code (e.g. a large number of local variables or a few distinct groups).  
+A local variable can also aide readability when using remote state files, for example:
+```
+locals {
+  vpn_cidrs = values(data.terraform_remote_state.network_common_infra.outputs.vpn_cidrs)
+}
+```
+Allows you to use the shortened form of `local.vpn_cidrs` throughout the module instead of `values(data.terraform_re...`.
```

Comment:
I'd rephrase to `This allows you to use a shorter and more contextual...`

Reply:
Updated as per comment. 👍